### PR TITLE
Add Generic F042C6

### PR DESCRIPTION
**Pull request by seeers**
Add support for STM32f042c6t6 
Good for USB - CAN Projects

### CODE REVIEW

**Comment by fpistm** [comment]
Hi @seeers 
you can use the python astyle script available in the CI/astyle to fix all issues.

https://github.com/stm32duino/wiki/wiki/Astyle

File: variants/Generic_F042Cx/variant.h [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+
+#ifdef __cplusplus
+extern "C" {
+#endif // __cplusplus
+
+/*----------------------------------------------------------------------------
+ *        Pins
+ *----------------------------------------------------------------------------*/
+
+#define PC13 0
```
Comment (fpistm) [inline variants/Generic_F042Cx/variant.h:25]: For generic, you should define the pin in the GPIO port ordering:

PA0
PA1
...

File: boards.txt [hunk 2]
```diff
@@ -833,6 +833,14 @@ GenF0.menu.pnum.DEMO_F030F4_HSI.build.board=DEMO_F030F4_HSI
 GenF0.menu.pnum.DEMO_F030F4_HSI.build.product_line=STM32F030x6
 GenF0.menu.pnum.DEMO_F030F4_HSI.build.variant=DEMO_F030F4
 
+# Generic_F042Cx
+GenF0.menu.pnum.Gen_F042Cx=Generic_F042C6 (internal RC oscillator)
```
Comment (fpistm) [inline boards.txt:837]: It should be fine to provide a system clock config to test several configs as done in GenF4.
HSE, HSE bypass, HSI.

File: variants/Generic_F042Cx/variant.h [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+#define PB4  29
+#define PB5  30
+#define PB6  31
+#define PB7  32
+#define PF11 33
+#define PB8  34
+#define PB9  35
+
+// This must be a literal
+// It is used with preprocessor tests (e.g. #if NUM_DIGITAL_PINS > 3)
```
Comment (fpistm) [inline variants/Generic_F042Cx/variant.h:63]: You can remove all useless comment.

**Comment by seeers** [comment]
> Hi @seeers
> you can use the python astyle script available in the CI/astyle to fix all issues.
> 
> https://github.com/stm32duino/wiki/wiki/Astyle

Thank you. I will use the script before the next commit.

File: boards.txt (hunk 2 above)
Comment (seeers) [reply boards.txt:837]: I'm not sure how to do it

**Comment by seeers** [comment]
Hi @fpistm 
could you tell me if the core compilation error caused by my commit?

**Comment by fpistm** [comment]
> 
> 
> Hi @fpistm
> could you tell me if the core compilation error caused by my commit?

Hi @seeers 
No it seems the new arduino-cli has some breaking changes in the last release. I'm currently investigate this.

**Comment by fpistm** [comment]
The cli has been fixed now (#1107)
Could you rebase your PR on top of the master then it can execute properly.
Thanks.

**Comment by fpistm** [comment]
Will be added later when #1091 will be done

**Comment by fpistm** [comment]
Hi @seeers 
I've included your PR in #1091, it has been reformatted to the new variant definition way. I've made some fixes.
So I close this one as it is superseded by #1091.
Thanks again for your contribution.

**Pull request closed without merging**